src/dmd/cparse.d

```diff
@@ -4557,6 +4571,35 @@ final class CParser(AST) : Parser!AST
         return s;
     }
 
+    /********************************
+     * Determines if type t is a function type.
+     * Make this work without needing semantic analysis.
+     * Params:
+     *  t = type to test
+     * Returns:
+     *  true if it represents a function
+     */
+    bool isfunction(AST.Type t)
+    {
+        //printf("isfunction() %s\n", t.toChars());
+        if (t.isTypeFunction())
+            return true;
+        if (auto tid = t.isTypeIdentifier())
+        {
+            /* Scan array of typedef identifiers that are an alias for
+             * a function type
+             */
+            foreach (ftid; funcTypeIds[])
```

Comment:
Not great that this is linear, so we're hoping that no one is going to have thousands of these typedefs in a library.

Reply:
They'd have to be thousands of _function_ typedefs, not just typedefs! Anyhow, I wanted this PR to be as simple as I could. I wanted to prove my idea for fixing this problem was workable, as it took me a couple days to come up with it. If we need a more sophisticated symbol table, that should be separate.

Reply:
Actually, keeping all typedefs in scope in a symbol table might help us with some of the ugly ambiguous cases that we try to patch up in the semantic.  But this is fine as-is for now.